components/experimental/src/duration/options.rs

```diff
@@ -0,0 +1,327 @@
+    /// Short style (default)
+    Short,
+    /// Long style (most verbose)
+    Long,
+    /// Digital formatting
+    Digital,
+}
+
+/// Configures the style of the year field.
+#[non_exhaustive]
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum YearStyle {
+    #[default]
+    /// The default style for this unit. Determined by base style.
+    Default,
+    /// Narrow style (most compact)
+    Narrow,
+    /// Short style (default)
+    Short,
+    /// Long style (most verbose)
+    Long,
+}
+
+/// Configures the style of the month field.
+#[non_exhaustive]
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum MonthStyle {
+    #[default]
+    /// The default style for this unit. Determined by base style.
+    Default,
+    /// Narrow style (most compact)
+    Narrow,
+    /// Short style (default)
+    Short,
+    /// Long style (most verbose)
+    Long,
+}
+
+/// Configures the style of the week field.
+#[non_exhaustive]
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum WeekStyle {
+    #[default]
+    /// The default style for this unit. Determined by base style.
+    Default,
+    /// Narrow style (most compact)
+    Narrow,
+    /// Short style (default)
+    Short,
+    /// Long style (most verbose)
+    Long,
+}
+
+/// Configures the style of the day field.
+#[non_exhaustive]
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum DayStyle {
+    #[default]
+    /// The default style for this unit. Determined by base style.
+    Default,
+    /// Narrow style (most compact)
+    Narrow,
+    /// Short style (default)
+    Short,
+    /// Long style (most verbose)
+    Long,
+}
+
+/// Configures the style of the hour field.
+#[non_exhaustive]
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum HourStyle {
+    #[default]
+    /// The default style for this unit. Determined by base style.
+    Default,
+    /// Narrow style (most compact)
+    Narrow,
+    /// Short style (default)
+    Short,
+    /// Long style (most verbose)
+    Long,
+    /// Ensure formatted value is at least two digits long (by appending leading zeroes, if necessary)
+    TwoDigit,
+    /// Numeric style
+    Numeric,
+}
+
+/// Configures the style of the minute field.
+#[non_exhaustive]
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum MinuteStyle {
+    #[default]
+    /// The default style for this unit. Determined by base style.
+    Default,
+    /// Narrow style (most compact)
+    Narrow,
+    /// Short style (default)
+    Short,
+    /// Long style (most verbose)
+    Long,
+    /// Ensure formatted value is at least two digits long (by appending leading zeroes, if necessary)
+    TwoDigit,
+    /// Numeric style
+    Numeric,
+}
+
+/// Configures the style of the second field.
+#[non_exhaustive]
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum SecondStyle {
+    #[default]
+    /// The default style for this unit. Determined by base style.
+    Default,
```

Comment:
This is can be moved to the code logic or we can call it `BaseStyle`

Reply:
Changed `Style::Default` to `Option<Style>`.